packages/office-ui-fabric-react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -53,53 +53,82 @@ export function initializeResponsiveMode(element?: HTMLElement): void {
 export function withResponsiveMode<TProps extends { responsiveMode?: ResponsiveMode }, TState>(
   ComposedComponent: new (props: TProps, ...args: any[]) => React.Component<TProps, TState>,
 ): any {
-  const resultClass = class WithResponsiveMode extends BaseDecorator<TProps, IWithResponsiveModeState> {
-    private _events: EventGroup;
+  // tslint:disable-next-line:function-name
+  function WithResponsiveModeBase(
+    props: TProps,
+    forwardedRef: React.Ref<React.Component<TProps, TState>>
+  ): JSX.Element | null {
+    const [responsiveMode = ResponsiveMode.unknown, setCurrentResponsiveMode] = React.useState<ResponsiveMode>(
+      _defaultMode || _lastMode || ResponsiveMode.large
+    );
+
+    const setCurrentResponsiveModeRef = React.useRef<typeof setCurrentResponsiveMode>();
+    setCurrentResponsiveModeRef.current = setCurrentResponsiveMode;
+
+    React.useEffect(() => {
+      return () => {
+        setCurrentResponsiveModeRef.current = undefined;
```

Comment:
what is this about? `layoutEffect` runs before `useEffect`, if you set this back to undefined, isn't this making `onResize` 
no-op after first run?

Reply:
This is only ever set to `undefined` when the whole component is unmounted.

Reply:
ah right. it's in return statement :)